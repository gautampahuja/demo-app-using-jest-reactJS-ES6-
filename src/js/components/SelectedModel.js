import React from 'react';

class SelectedModel extends React.Component {
    render () {
        var selectedCarName = this.props.selectedCarName;
        return (
            <div className="searchResults">
                {
                    selectedCarName != undefined
                        ?
                        <h4><u>Selected Model</u>: {selectedCarName}</h4>
                        :
                        ""
                }
            </div>
        )
    }
}
export default SelectedModel;