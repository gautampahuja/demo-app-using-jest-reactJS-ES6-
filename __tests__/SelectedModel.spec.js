/* global describe, test, expect */

import React from 'react'
import SelectedModel from '../src/js/components/SelectedModel.js'
import renderer from 'react-test-renderer'

describe('SelectedModel component', () => {
  test('SelectedModel is rendered without selectedCarName object', () => {
    const component = renderer.create(<SelectedModel/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot()
  });

  test('SelectedModel is rendered with selectedCarName object ', () => {
    const selectedCarName = "MERCEDES BENZ";
    const component = renderer.create(<SelectedModel selectedCarName={selectedCarName} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot()
  })
});
