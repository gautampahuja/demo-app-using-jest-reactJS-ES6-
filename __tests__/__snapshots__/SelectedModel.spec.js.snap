// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedModel component SelectedModel is rendered with selectedCarName object  1`] = `
<div
  className="searchResults"
>
  <h4>
    <u>
      Selected Model
    </u>
    : 
    MERCEDES BENZ
  </h4>
</div>
`;

exports[`SelectedModel component SelectedModel is rendered without selectedCarName object 1`] = `
<div
  className="searchResults"
>
  
</div>
`;
